tests/test_pipeline_invariants_v2.py

```
@@ -0,0 +1,174 @@
+    first = ["T2: статья 469 ГК РК: пересказ обобщает узкое условие нормы"]
+    second = ["FINAL_RELEASE_CITATION: статья 469 ГК РК: пересказ обобщает узкое условие нормы"]
+
+    assert _semantic_issues(first) == _semantic_issues(second)
+
+
+class _FakeState:
+    def __init__(self) -> None:
+        self.key = SimpleNamespace(bot_id=1, chat_id=10, user_id=20, thread_id=None)
+        self._data: dict[str, object] = {}
+
+    async def get_data(self):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `get_data`

(ANN202)